nnvm/python/nnvm/testing/gradient.py

```diff
@@ -0,0 +1,179 @@
+"""Helper utilities to test gradients."""
+from __future__ import absolute_import as _abs
+
+import numpy as np
+import tvm
+from tvm.contrib import graph_runtime
+from ..compiler import graph_util
+from ..import compiler
+from ..import symbol
+from ..import graph
+
+def check_gradients_numeric(y, inputs=None, dtype='float32', print_graph=False,
```

Comment:
Do we really need to implement another gradient test interface? Since there is backward interface to test operator gradient, can we reuse that interface?

Reply:
Do you mean the `helper` function from `test_top_level1.py`? Yes, I think it's a good idea to move this helper function into nnvm.testing and improve it with the ability to compute gradients numerically. I'll try to do this in a couple of days.

Reply:
Yes. That would be great.